app/assets/javascripts/discourse/app/static/wizard/components/fields/radio.gjs

```diff
@@ -0,0 +1,75 @@
+import Component from "@glimmer/component";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action, set } from "@ember/object";
+import { hash } from "@ember/helper";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class Radio extends Component {
+  constructor() {
+    super(...arguments);
+
+    this._setSelected();
+  }
+
+  get field() {
+    return this.args.field;
+  }
+
+  @action
+  selectionChanged(input) {
+    this.field.value = input.target.value;
+    this._resetSelected();
+    this._setSelected();
+  }
+
+  _resetSelected() {
```

Comment:
I think you can get rid of this function if you do `set(choice, "selected", false);` in the negative branch of `if (this.field.value === choice.id) {` in the other function.

Reply:
Nice one! 👌